addons/mail/static/src/messaging/component/notification_list/notification_list_notification_group_tests.js

```diff
@@ -0,0 +1,629 @@
+'use strict';
+
+const components = {
+    NotificationList: require('mail.messaging.component.NotificationList'),
+};
+
+const {
+    afterEach: utilsAfterEach,
+    afterNextRender,
+    beforeEach: utilsBeforeEach,
+    pause,
+    start: utilsStart,
+} = require('mail.messaging.testUtils');
+
+QUnit.module('mail', {}, function () {
+QUnit.module('messaging', {}, function () {
+QUnit.module('component', {}, function () {
+QUnit.module('NotificationList_NotificationGroup', {
+    beforeEach() {
+        utilsBeforeEach(this);
+
+        /**
+         * @param {Object} param0
+         * @param {string} [param0.filter='all']
+         */
+        this.createNotificationListComponent = async ({ filter = 'all' } = {}) => {
+            const NotificationListComponent = components.NotificationList;
+            NotificationListComponent.env = this.env;
+            this.component = new NotificationListComponent(null, { filter });
+            await afterNextRender(async () => {
+                await this.component.mount(this.widget.el);
+            });
+        };
+
+        this.start = async params => {
+            let { env, widget } = await utilsStart(Object.assign({}, params, {
+                data: this.data,
+            }));
+            this.env = env;
+            this.widget = widget;
+        };
+    },
+    afterEach() {
+        utilsAfterEach(this);
+        if (this.component) {
+            this.component.destroy();
+            this.component = undefined;
+        }
+        if (this.widget) {
+            this.widget.destroy();
+            this.widget = undefined;
+        }
+        this.env = undefined;
+        delete components.NotificationList.env;
+    },
+});
+
+QUnit.test('notification group basic layout', async function (assert) {
+    assert.expect(10);
+
+    this.data.initMessaging.mail_failures = {
+        11: {
+            last_message_date: moment.utc().format("YYYY-MM-DD HH:mm:ss"),
+            message_id: 11,
+            message_type: 'email',
+            model: 'mail.channel',
+            model_name: "Channel",
+            notifications: {
+                21: {
+                    failure_type: 'SMTP',
+                    notification_id: 21,
+                    notification_status: 'exception',
+                    notification_type: 'email',
+                    partner_id: 41,
+                    partner_name: "Someone",
+                    res_id: 31,
+                },
+            },
+            res_id: 31,
+        },
+    };
+
+    await this.start();
+    await this.createNotificationListComponent();
+
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationGroup',
+        "should have 1 notification group"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationGroup_name',
+        "should have 1 group name"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_NotificationGroup_name').innerText,
```

Comment:
any reason to use `innerText` over `textContent`? Don't think textcontent has some style that we want to take into account, and `innerText` triggers a reflow so prefer `textContent` instead (check whole diff)

Reply:
No specific reason, I didn't really pay attention and it worked like this.

I'll have to add some spaces here and there in the expected result with `textContent`, but I'll do it.